lua/weapons/weapon_tag.lua

```diff
@@ -0,0 +1,150 @@
+
+AddCSLuaFile()
+
+SWEP.PrintName = "Tag"
+SWEP.Author = "Ugleh"
+SWEP.Purpose = "Tag someone within 30 seconds or you will die."
+SWEP.Slot = 0
+SWEP.SlotPos = 4
+SWEP.Weight					= 99
+SWEP.AutoSwitchTo			= true
+SWEP.AutoSwitchFrom			= false
+
+SWEP.Spawnable = true
+
+SWEP.ViewModel = Model( "models/weapons/c_arms.mdl" )
+SWEP.WorldModel = ""
+SWEP.ViewModelFOV = 54
+SWEP.UseHands = true
+
+SWEP.Primary.ClipSize = -1
+SWEP.Primary.DefaultClip = -1
+SWEP.Primary.Automatic = true
+SWEP.Primary.Ammo = "none"
+
+SWEP.Secondary.ClipSize = -1
+SWEP.Secondary.DefaultClip = -1
+SWEP.Secondary.Automatic = true
+SWEP.Secondary.Ammo = "none"
+
+SWEP.DrawAmmo = false
+
+SWEP.HitDistance = 48
+
+
+local justReloaded = 0
+local initializeTime = 0
+local tagTimer = 35
+function SWEP:Initialize()
+	self:SetHoldType( "fist" )
+	initializeTime = CurTime()
+	if SERVER then
+		timer.Simple(5,function() if IsValid(self) and self.Owner:IsFrozen() then self.Owner:UnLock() end end)
+		timer.Simple(tagTimer,function() if IsValid(self) then self:Remove() self.Owner:Kill() end end)
+	end
+end
+
+function SWEP:Deploy()
+	self:SetHoldType("fist")
+	local vm = self.Owner:GetViewModel()
+	vm:SendViewModelMatchingSequence( vm:LookupSequence( "fists_draw" ) )
+
+end
+
+function SWEP:Reload()
+	if justReloaded<1 then
+		local pitch = 100 + (self.Owner:Crouching() and 40 or 0)
+		self:ExtEmitSound("tag/dead.wav", {speech=0.8, volume=0.35, pitch=pitch, shared=true})
+	end
+	justReloaded=2
+end
+
+function SWEP:Tick()
+	justReloaded = justReloaded-1
+end
+
+
+function TagPlayer(self,target,attacker)
```

Comment:
This func name seems like it could cause a conflict, can you make it local please?

Reply:
I assumed everything encapsulated in the class was already local to the class itself. My bad.

Reply:
Nah it's global. You could do SWEP:TagPlayer(target,attacker) though, that would be better anyway